### PR TITLE
[Feature] Add SaveMode support for hive connector

### PR DESCRIPTION
**Pull request by liunaijie**
### Purpose of this pull request

subtask of #5390 

### Check list

* [x] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by chl-wxp** [comment]
Hi,Please modify the corresponding documentation

**Review by hailin0 — COMMENTED** [review]
Please add testcase or e2e

File: seatunnel-connectors-v2/connector-hive/src/main/java/org/apache/seatunnel/connectors/seatunnel/hive/sink/HiveSink.java [hunk 1]
```diff
@@ -207,4 +71,21 @@ public void prepare(Config pluginConfig) throws PrepareFailException {
...
+        if (pluginConfig.hasPath(SCHEMA_SAVE_MODE.key())
+                || pluginConfig.hasPath(DATA_SAVE_MODE.key())) {
+            ReadonlyConfig readonlyConfig = ReadonlyConfig.fromConfig(pluginConfig);
+            SchemaSaveMode schemaSaveMode = readonlyConfig.get(SCHEMA_SAVE_MODE);
+            DataSaveMode dataSaveMode = readonlyConfig.get(DATA_SAVE_MODE);
+            String custom_sql = readonlyConfig.get(CUSTOM_SQL);
+            new ReadonlyConfigParser().parse(readonlyConfig);
+            HiveCatalog catalog = new HiveCatalog(pluginConfig);
+            return Optional.of(
+                    new DefaultSaveModeHandler(
```
Comment (EricJoy2048) [inline seatunnel-connectors-v2/connector-hive/src/main/java/org/apache/seatunnel/connectors/seatunnel/hive/sink/HiveSink.java:86]: When people use Hive to automatically create tables, they may need to specify location, field delimiters, storage formats, etc. So I think Hive's automatic table creation should mimic Doris by specifying a table creation template for creation, and by modifying the template to specify table properties when needed.

**Comment by dailai** [comment]
@liunaijie Hi, Do you still have time to follow up this pr? If not, I can continue to do it

**Comment by liunaijie** [comment]
> @liunaijie Hi, Do you still have time to follow up this pr? If not, I can continue to do it

hi, i am working on it.

**Comment by dailai** [comment]
> > @liunaijie Hi, Do you still have time to follow up this pr? If not, I can continue to do it
> 
> hi, i am working on it.

Okay, thank you.

**Comment by liunaijie** [comment]
will submit this feature by another pr

**Pull request closed without merging**